Add mode-of-transport question

wielandb opened the pull request:
This PR adds a question that asks what kind of vehicles stop at a platform. It only asks the question if there are no indicators whatsoever about what vehicles stop at the platform.

7h30n3 commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+                  }
+              },
+              {
+                  "name": "Citiybahn",
```
We need to find something else. "City-Bahn" is not well known. Let's call it "Stadtbahn". And add a short description based on this article: https://wiki.openstreetmap.org/wiki/DE:Tag:railway%3Dlight_rail

7h30n3 commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+      "text": "Welche(s) Verkehrsmittel halten an dieser Haltestelle?"
+      },
+      "answer": {
+          "type": "MultiList",
```
Add an image to each answer.

7h30n3 commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+                      "ferry": false
+                  },
+                  "!osm_tags":{
+                      "highway":"bus_stop",
```
Add missing space.

7h30n3 commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+        "conditions": [
+              {
+                  "osm_tags": {
+                      "public_transport":"platform",
```
Add missing space.

7h30n3 commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+                      "tram": false,
+                      "light_rail": false,
+                      "monorail": false,
+                      "train":false,
```
Add missing space.

Robbendebiene commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+                  "osm_tags": {
+                      "public_transport":"platform",
+                      "bus": false,
+                      "tram": false,
```
Add `"subway": false`

Robbendebiene commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+                  "name": "Citiybahn",
+                  "osm_tags": {
+                      "light_rail": "yes"
+              }
```
Add answer for subway (U-Bahn) and monorail (Einschienenbahnen).
monorail should be the last entry in the list.

Robbendebiene commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+                      "ferry": false
+                  },
+                  "!osm_tags":{
+                      "highway":"bus_stop",
+                      "amenity": "bus_station"
+                  }
```
I guess this is used to exclude objects where either `"highway":"bus_stop",` OR `"amenity": "bus_station"` is set. But it actually only excludes elements where `"highway":"bus_stop",` AND `"amenity": "bus_station"` is set.

I prefer changing this to the following condition (we use this to exclude questions for masts when there is a platform, if there is no platform we ask the questions for the mast)

```
            "!parent": [
               {
                  "osm_tags": {
                     "public_transport": "platform"
                  }
               }
            ]
```


____________

I just noticed, that we ran into the same misconception regarding https://github.com/OPENER-next/OpenStop/pull/178 Currently this can only be solved by "duplicating" permissions. Maybe for now we could remove the `foot` condition. Let's discuss this in our weekly meeting.

```
          "!osm_tags": {
               "access": ["no","private"],
               "foot": ["no", "private"]
            },
```

wielandb replied:
Hm.... this was not what I meant, actually. 😮
My idea was that something that is tagged with `highway=bus_stop` is undoubtly a bus stop and we should not ask for the vehicles. But of course, as I think of it, it is possible that a `highway=bus_stop` means a bus stop where also trams depart. So we should still ask the question, even if it is set....

I added the new condition.

wielandb commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,64 @@
+              }
+        ]
    }
 ]
```
I hope it is corrected now.

wielandb commented:
I added conditions so that the three cases where the question is asked are:
- `public_transport=platform` without any vehicles given
- `railway=platform` without any vehicles given
- `highway=bus_stop` without any vehciles given, who also does not have `public_transport=platform` as a parent

Robbendebiene commented on assets/datasets/question_catalog.json:
```diff
@@ -2655,5 +2655,108 @@
+               }
+            ],
+         "constructor": {
+            "bus": ["COALESCE", "$input"],
+            "tram": ["COALESCE", "$input"],
+            "train": ["COALESCE", "$input"],
+            "light_rail": ["COALESCE", "$input"]
```
still misses `monorail` and `subway`

7h30n3 commented:
Thanks for adding images.

* Please replace the following images with these:
tram: https://commons.wikimedia.org/wiki/File:Moscow,_Zagorodnoe_Highway_tram_stop_June_2022_03.jpg
subway: https://commons.wikimedia.org/wiki/File:Kyoto_subway_2831_Matsugasaki_20220506.jpg
bus: https://commons.wikimedia.org/wiki/File:20190918_Oxford_Bus_841.jpg

* Is there a particular reason why you use the png file format for photographs? If not please use jpg.
* The shortest side of an image needs to be (at least) 2000px and the file size usually should be between 1 and 2 MB, please take this into consideration.
* If you need to crop an image please keep the aspect ratio of the original (often 5:4) or use 1:1
* Also please remove the prefix `mode_of_transportation_` in the filenames.

wielandb commented:
> * The shortest side of an image needs to be (at least) 2000px and the file size usually should be between 1 and 2 MB, please take this into consideration.

Done. I was orienting myself at the suggestions made [here](https://github.com/OPENER-next/OpenStop/pull/168#pullrequestreview-1406057177)
I updated the pictures where possible so that the shortest side is always at least 2000px. I choose quality settings that made every image (where possible) approx. 1.5 Mb in size.

> * If you need to crop an image please keep the aspect ratio of the original (often 5:4) or use 1:1

Changed. I thought it would be more important to have a consistent aspect ratio between all images of a question than to not change it in regards to the original.

wielandb commented:
Should be ready for re-review.

7h30n3 commented:
Thanks for updating the PR.

> > * The shortest side of an image needs to be (at least) 2000px and the file size usually should be between 1 and 2 MB, please take this into consideration.
> 
> Done. I was orienting myself at the suggestions made [here](https://github.com/OPENER-next/OpenStop/pull/168#pullrequestreview-1406057177) I updated the pictures where possible so that the shortest side is always at least 2000px. I choose quality settings that made every image (where possible) approx. 1.5 Mb in size.

Sorry, my bad. All images I contribute had a side length of 2000px. So the "at least" in my first comment was somewhat contradicting to your [reference](https://github.com/OPENER-next/OpenStop/pull/168#pullrequestreview-1406057177).

> > * If you need to crop an image please keep the aspect ratio of the original (often 5:4) or use 1:1
> 
> Changed. I thought it would be more important to have a consistent aspect ratio between all images of a question than to not change it in regards to the original.

That's right, but why choosing 16 by 9 and not keeping the original ones. As most image sensors have a very similar aspect ration anyway.
